rest/model/task_annotations.go

```diff
@@ -203,3 +250,18 @@ func ValidateIssues(issues []APIIssueLink) error {
 	}
 	return catcher.Resolve()
 }
+
+func ValidateTaskLinks(links []APITaskLink) error {
+	settings := evergreen.GetEnvironment().Settings()
+	catcher := grip.NewBasicCatcher()
+	for _, link := range links {
+		catcher.Add(util.CheckURL(utility.FromStringPtr(link.URL)))
+		if !utility.StringMatchesAnyRegex(utility.FromStringPtr(link.URL), settings.Ui.CORSOrigins) {
+			catcher.Errorf("task link URL '%s' must match a CORS origin", utility.FromStringPtr(link.URL))
+		}
+		if utility.FromStringPtr(link.Text) == "" {
```

Comment:
Can we validate that this isn't too long?

Reply:
Sure, setting the limit at 1024 characters initially but fine to restrict it further if needed.

Reply:
I would vote to restrict it to only 40 characters.
<img width="309" alt="image" src="https://user-images.githubusercontent.com/4605522/228857461-630c9a5a-724e-4f72-9c92-25d66b2b180e.png">